backend/data/users.js

```diff
@@ -308,20 +365,84 @@ async function addGrade(studentId, courseId, assignmentId, grade) {
  * @returns object
  */
 async function fetchGrade(assignmentId, studentId) {
-  const errorMSg = 'No assignments found for the given assignment id';
+  const parsedAssignmentId = error.validId(assignmentId);
   const user = await getUser(studentId);
+  if (!user) throw new Error('No user was found with the given id');
+  const errorMSg = 'No assignments found for the given assignment id';
+
   if (user && !user.classes) throw new Error(errorMSg);
   for (const classObj of user.classes) {
-    const assignment = classObj.grades.find((e) => e._id === assignmentId);
+    const assignment = classObj.grades.find((e) => e._id.toString() === assignmentId);
     if (!assignment) throw new Error(errorMSg);
     const response = {
-      status: "success",
-      grade: assignment.grade
+      "status": "success",
+      "grade": assignment.grade
     }
     return response;
   }
 }
 
+/**
+ * Fetch all grades for the assignment
+ * @param {string} assignmentId 
+ * @returns array
+ */
+async function fetchAllGrades(assignmentId) {
+  const userCollection = await users();
+  const parsedAssignmentId = error.validId(assignmentId);
+  const grades = await userCollection.aggregate([
+    { $match: { 'classes': { $elemMatch: { 'grades': { $elemMatch: { _id: parsedAssignmentId } } } } } },
+    { $unwind: "$classes" },
+    { $unwind: "$classes" },
+    { $unwind: "$classes.grades" },
+    { $unwind: "$classes.grades" },
+    { $match: { 'classes.grades._id': parsedAssignmentId } },
+    { $project: { _id: 0, firstName: 1, lastName: 1, grade: "$classes.grades.grade" } }
+  ]).toArray();
+  if (grades.length === 0) throw new Error('No grades found for the assignment');
+  return grades;
```

Comment:
Should probably have this return a response along the lines of:
const response = {
    "status": "success",
    "grades": grades
}
just to keep consistent with other functions/routes

Reply:
I have made changes to just return a response as above, this will be reflected in the latest PR.